pydantic/_internal/_generate_schema.py

```diff
@@ -1311,22 +1309,22 @@ def _tuple_schema(self, tuple_type: Any) -> core_schema.CoreSchema:
         # This is only true for <3.11, on Python 3.11+ `typing.Tuple[()]` gives `params=()`
         if not params:
             if tuple_type in TUPLE_TYPES:
-                return core_schema.tuple_variable_schema()
+                return core_schema.tuple_schema([core_schema.any_schema()], variadic_item_index=0)
             else:
                 # special case for `tuple[()]` which means `tuple[]` - an empty tuple
-                return core_schema.tuple_positional_schema([])
+                return core_schema.tuple_schema([])
         elif params[-1] is Ellipsis:
             if len(params) == 2:
-                return self._tuple_variable_schema(tuple_type, params[0])
+                return core_schema.tuple_schema([self.generate_schema(params[0])], variadic_item_index=0)
             else:
                 # TODO: something like https://github.com/pydantic/pydantic/issues/5952
                 raise ValueError('Variable tuples can only have one type')
```

Comment:
@dmontagu I suspect that this is where the new variadic tuple functionality can be extended as per #5952. I'll defer this to a future PR in your hands.